linux/deploy_vm/ubuntu/get_mainline_kernel.yml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2022-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Set the base url for kernel version check"
+  ansible.builtin.set_fact:
+    mainline_kernel_url: "https://kernel.ubuntu.com/mainline"
+
+- name: "Check the kernel version exists"
+  ansible.builtin.uri:
+    url: "{{ mainline_kernel_url }}/v{{ upgrade_kernel_version }}/amd64"
+    return_content: true
+  register: check_kernel_result
+  ignore_errors: yes
```

Comment:
Please replace "yes" with "true". If this task is just to check whether the url exists, `return_content: true` can be removed since you didn't use its content at all.